File: backend/app/schemas/help.py
from sqlmodel import SQLModel


class HelpArticleSummary(SQLModel):
    slug: str
    title: str
    category: str
    summary: str | None = None
    is_faq: bool


class HelpArticleListResponse(SQLModel):
    items: list[HelpArticleSummary]
    top_faqs: list[HelpArticleSummary]


class HelpArticleDetail(SQLModel):
    slug: str
    title: str
    category: str
    summary: str | None = None
    is_faq: bool
    # Raw markdown — the frontend renders and sanitizes it before display.
    # This backend never renders HTML from article content.
    body_markdown: str

File: frontend/src/api/help.ts
import api from "@/api/client"

export interface HelpArticleSummary {
  slug: string
  title: string
  category: string
  summary: string | null
  is_faq: boolean
}

export interface HelpArticleListResponse {
  items: HelpArticleSummary[]
  /**
   * Populated **only** when a non-empty `search` matched nothing — the
   * empty-state fallback, not a "featured" list. Empty on a plain browse
   * and on a category-only filter, even when `items` is also empty.
   */
  top_faqs: HelpArticleSummary[]
}

export interface HelpArticleDetail extends HelpArticleSummary {
  /** Raw markdown. The backend never renders HTML — sanitize before display. */
  body_markdown: string
}

export interface GetHelpArticlesParams {
  search?: string
  category?: string
}

export async function getHelpArticles(params: GetHelpArticlesParams = {}) {
  const { data } = await api.get<HelpArticleListResponse>("/help/articles", { params })
  return data
}

/**
 * 404, never 403, when the slug exists but the caller's role can't see it —
 * the response must not confirm the article exists. Callers should render
 * the 404 message as-is rather than softening it into "you don't have
 * access to this".
 */
export async function getHelpArticle(slug: string) {
  const { data } = await api.get<HelpArticleDetail>(`/help/articles/${slug}`)
  return data
}
